api/apps/v1alpha1/nemo_datastorev2_types.go

```diff
@@ -0,0 +1,1046 @@
+func (n *NemoDatastoreV2) GetDefaultLivenessProbe() *corev1.Probe {
+	probe := corev1.Probe{
+		InitialDelaySeconds: 15,
+		TimeoutSeconds:      1,
+		PeriodSeconds:       10,
+		SuccessThreshold:    1,
+		FailureThreshold:    3,
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: "/v1/health/live",
+				Port: intstr.IntOrString{
+					Type:   intstr.Type(0),
+					IntVal: n.Spec.Expose.Service.Port,
+				},
+			},
+		},
+	}
+
+	return &probe
+}
+
+// GetReadinessProbe returns readiness probe for the NemoDatastoreV2 container
+func (n *NemoDatastoreV2) GetReadinessProbe() *corev1.Probe {
+	if n.Spec.ReadinessProbe.Probe == nil {
+		return n.GetDefaultReadinessProbe()
+	}
+	return n.Spec.ReadinessProbe.Probe
+}
+
+// GetDefaultReadinessProbe returns the default readiness probe for the NemoDatastoreV2 container
+func (n *NemoDatastoreV2) GetDefaultReadinessProbe() *corev1.Probe {
+	probe := corev1.Probe{
+		InitialDelaySeconds: 15,
+		TimeoutSeconds:      1,
+		PeriodSeconds:       10,
+		SuccessThreshold:    1,
+		FailureThreshold:    3,
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: "/v1/health/ready",
+				Port: intstr.IntOrString{
+					Type:   intstr.Type(0),
+					IntVal: n.Spec.Expose.Service.Port,
+				},
+			},
+		},
+	}
+
+	return &probe
+}
+
+// GetStartupProbe returns startup probe for the NemoDatastoreV2 container
+func (n *NemoDatastoreV2) GetStartupProbe() *corev1.Probe {
+	return n.Spec.StartupProbe.Probe
+}
+
+// GetDefaultStartupProbe returns the default startup probe for the NemoDatastoreV2 container
+func (n *NemoDatastoreV2) GetDefaultStartupProbe() *corev1.Probe {
+	probe := corev1.Probe{
+		InitialDelaySeconds: 40,
+		TimeoutSeconds:      1,
+		PeriodSeconds:       10,
+		SuccessThreshold:    1,
+		FailureThreshold:    180,
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: "/v1/health/ready",
+				Port: intstr.IntOrString{
+					Type:   intstr.Type(0),
+					IntVal: n.Spec.Expose.Service.Port,
+				},
+			},
+		},
+	}
+
+	return &probe
+}
+
+// GetVolumeMounts returns volumes for the NemoDatastoreV2 container
+func (n *NemoDatastoreV2) GetVolumeMounts() []corev1.VolumeMount {
+	return []corev1.VolumeMount{
+		{
+			MountPath: "/tmp",
+			Name:      "temp",
+		},
+		{
+			MountPath: "/data",
+			Name:      "data",
+		},
+	}
+}
+
+func (n *NemoDatastoreV2) GetVolumeMountsInitContainer() []corev1.VolumeMount {
+	return []corev1.VolumeMount{
+		{
+			MountPath: "/usr/sbin",
+			Name:      "config",
+		},
+		{
+			MountPath: "/tmp",
+			Name:      "temp",
+		},
+		{
+			MountPath: "/data",
+			Name:      "data",
+		},
+		{
+			MountPath: "/env-to-ini-mounts/inlines/",
+			Name:      "inline-config-sources",
+		},
+		{
+			MountPath: "/usr/sbin/init",
+			Name:      "init",
+		},
+	}
+}
+
+func (n *NemoDatastoreV2) GetInitContainers() []corev1.Container {
```

Comment:
currently we are not rendering initContainers in the deployment/statefulset manifests, we need to add that logic.

Reply:
See `internal/controller/nemo_datastorev2_controller.go:L382`